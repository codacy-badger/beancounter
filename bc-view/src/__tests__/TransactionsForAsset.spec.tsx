import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import nock from "nock";
import { MemoryRouter } from "react-router";
import Trades from "../trns/Trades";

afterEach(cleanup);

const bff = "http://localhost";
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key })
}));

nock(bff, {
  reqheaders: {
    authorization: "Bearer undefined"
  }
})
  .get("/bff/trns/test/asset/alphabet/trades")
  .replyWithFile(200, __dirname + "/__contracts__/trans-for-asset.json", {
    "Access-Control-Allow-Origin": "*",
    "Content-type": "application/json"
  });

nock(bff, {
  reqheaders: {
    authorization: "Bearer undefined",
  },
})
  .get("/bff/assets/alphabet")
  .replyWithFile(200, __dirname + "/__contracts__/alphabet.json", {
    "Access-Control-Allow-Origin": "*",
    "Content-type": "application/json",
  });

describe("<Trades />", () => {
  it("trades should match snapshot", async () => {
    const TestTrnForAsset = (): JSX.Element => {
      return Trades("test", "alphabet");
    };

    const { container } = render(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <TestTrnForAsset />
      </MemoryRouter>
    );
    await screen.findByText("BUY");
    expect(nock.isDone());
    expect(container).toMatchSnapshot();
  });
});
