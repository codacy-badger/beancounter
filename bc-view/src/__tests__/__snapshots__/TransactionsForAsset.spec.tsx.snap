// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transactions /> should match snapshot 1`] = `
<div>
  <div>
    <nav
      class="container"
    >
      <div
        class="page-title"
      >
        <div
          class="column page-title subtitle is-6"
        >
          Alphabet Inc - Ordinary Shares - Class C
          :
          NASDAQ
        </div>
      </div>
    </nav>
    <div
      class="page-box is-primary has-background-light"
    >
      <div
        class="container"
      >
        <table
          class="table is-striped is-hoverable"
        >
          <thead>
            <tr>
              <th>
                Type
              </th>
              <th>
                Currency
              </th>
              <th>
                Trade Date
              </th>
              <th
                align="right"
              >
                Quantity
              </th>
              <th
                align="right"
              >
                Price
              </th>
              <th
                align="right"
              >
                Amount
              </th>
              <th>
                Action
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                BUY
              </td>
              <td>
                USD
              </td>
              <td>
                2019-01-18
              </td>
              <td
                align="right"
              >
                <span>
                  2
                </span>
              </td>
              <td
                align="right"
              >
                <span>
                  1,080.00
                </span>
              </td>
              <td
                align="right"
              >
                <span>
                  2,160.01
                </span>
              </td>
              <td>
                <a
                  class="fa fa-edit"
                  href="/trns/SHEETS/0/68"
                />
                <span>
                   
                </span>
                <a
                  class="fa fa-remove has-padding-left-6"
                  href="/portfolio/delete/[object Object]"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
